api_tests/tokens/views/test_token_detail.py

```diff
@@ -1,219 +1,203 @@
-                                          'id': self.user1_token._id,
-                                          'type': 'tokens'}}, auth=self.user1.auth)
-        assert_equal(res.status_code, 200)
-
-        list_url = TOKEN_LIST_URL
-        res = self.app.get(list_url, auth=self.user1.auth)
-        assert_equal(res.status_code, 200)
-        assert_equal(len(res.json['data']),
-                     1)
+        new_name = 'The token formerly known as Prince'
+        res = app.patch_json_api(url_token_detail,
+                                    {'data': {
+                                        'attributes': {'name': new_name, 'scopes':'osf.full_write'},
+                                        'id': token_user_one._id,
+                                        'type': 'tokens'}}, auth=user_one.auth)
+        assert res.status_code == 200
+
+        res = app.get(url_token_list, auth=user_one.auth)
+        assert res.status_code == 200
+        assert (len(res.json['data']) == 1)
 
     @mock.patch('framework.auth.cas.CasClient.revoke_tokens')
-    def test_deleting_token_flags_instance_inactive(self, mock_method):
+    def test_deleting_token_flags_instance_inactive(self, mock_method, app, url_token_detail, user_one, token_user_one):
         mock_method.return_value(True)
-        res = self.app.delete(self.user1_token_url, auth=self.user1.auth)
-        self.user1_token.reload()
-        assert_false(self.user1_token.is_active)
-
-    def test_read_does_not_return_token_id(self):
-        res = self.app.get(self.user1_token_url, auth=self.user1.auth)
-        assert_equal(res.status_code, 200)
-        assert_false(res.json['data']['attributes'].has_key('token_id'))
-
-    def test_create_with_admin_scope_fails(self):
-        res = self.app.post_json_api(TOKEN_LIST_URL, self.injected_scope, auth=self.user1.auth, expect_errors=True)
-        assert_equal(res.status_code, 400)
-        
-    def test_create_with_fake_scope_fails(self):
-        res = self.app.post_json_api(TOKEN_LIST_URL, self.nonsense_scope, auth=self.user1.auth, expect_errors=True)
-        assert_equal(res.status_code, 400)
-
-    def test_update_with_admin_scope_fails(self):
-        res = self.app.put_json_api(self.user1_token_url, self.injected_scope, auth=self.user1.auth, expect_errors=True)
-        assert_equal(res.status_code, 400)
-
-    def test_update_with_fake_scope_fails(self):
-        res = self.app.put_json_api(self.user1_token_url, self.nonsense_scope, auth=self.user1.auth, expect_errors=True)
-        assert_equal(res.status_code, 400)
+        res = app.delete(url_token_detail, auth=user_one.auth)
+        token_user_one.reload()
+        assert not token_user_one.is_active
+
+    def test_read_does_not_return_token_id(self, app, url_token_detail, user_one):
+        res = app.get(url_token_detail, auth=user_one.auth)
+        assert res.status_code == 200
+        assert not res.json['data']['attributes'].has_key('token_id')
```

Comment:
I realize that this is translated from old code, but `in` is preferred to `has_key`. In fact, `has_key` has been removed in Python 3: https://docs.python.org/3.1/whatsnew/3.0.html#builtins